alexandria-reader/frontend
# feat: add text pagination and reading progress

## Description

Adds paginated text listing, in-text page navigation, and reading position persistence.

- Text list pagination with Next/Previous controls
- Text body split into pages via word-count tokenizer
- Reading position saved and restored via backend
- Dictionary lookup changed from inline iframe to popup window
- Added eslint no-restricted-syntax rules

Goes with https://github.com/alexandria-reader/backend/pull/164

## Related Issue

Fixes #288

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  x  | :bug: Bug fix              |
|  x  | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Testing Steps / QA Criteria

- Open texts page — Next/Previous buttons should paginate at 10 texts
- Open a text — page through with navigation controls
- Close and reopen a text — should resume at last page
- Click WordReference/DeepL buttons — should open popup windows
